src/coreclr/jit/importercalls.cpp

```diff
@@ -384,8 +384,18 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
                 // take the call now....
                 call = gtNewIndCallNode(nullptr, callRetTyp, di);
 
+                if (sig->isAsyncCall())
+                {
+                    impSetupAndSpillForAsyncCall(call->AsCall(), opcode, prefixFlags);
+                }
+
                 impPopCallArgs(sig, call->AsCall());
 
+                if (call->AsCall()->IsAsync())
+                {
+                    impInsertAsyncContinuationForLdvirtftnCall(call->AsCall());
+                }
+
```

Comment:
So the culprit was the `goto DONE;` below, thus `CORINFO_VIRTUALCALL_LDVIRTFTN` was not going through the codepath that eventually inserts continuation arg for other kinds of calls.